config/external-dns/deployment.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+  namespace: hive
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: external-dns
+  template:
+    metadata:
+      labels:
+        name: external-dns
+    spec:
+      strategy:
+        type: Recreate
+      serviceAccountName: external-dns
+      containers:
+      - name: external-dns
+        image: registry.svc.ci.openshift.org/openshift/hive-v4.0:external-dns
```

Comment:
Not sure how to handle this image reference. If we deploy this one in stage and prod, it's possible we will be unclear what we're running, deployments won't bounce if a new image is published because nothing has changed, and we might also inadvertently start running new code in prod that we haven't tested in master just by virtue of the pod restarting for some unknown reason.

Feels like we may need to publish our own images for this to pin them for time being. This probably parallels with the card I have in Jira to check in with ART team on what we can do for publishing images outside of the release.